neutron/services/loadbalancer/agent/agent_manager.py

```diff
@@ -332,3 +332,24 @@ def agent_updated(self, context, payload):
                              pool_id)
                     self._destroy_pool(pool_id)
             LOG.info(_("Agent_updated by server side %s!"), payload)
+
+    def add_pool(self, context, pool, driver_name):
```

Comment:
恩， remove的时候update pool status 为down，调用agent原有的delete
add的时候update pool status为PENDING_UPDATE,然后先调用agent原有的
create，在调用原有的update（这里要先调用create，否则直接调用update， 底层会有记录 认为该pool不存在）

2016-07-13 11:34 GMT+08:00 huntxu <notifications@github.com>:

> In neutron/services/loadbalancer/agent/agent_manager.py
> <https://github.com/eayunstack/neutron/pull/39#discussion_r70560261>:
>
> > @@ -332,3 +332,24 @@ def agent_updated(self, context, payload):
> >                               pool_id)
> >                      self._destroy_pool(pool_id)
> >              LOG.info(_("Agent_updated by server side %s!"), payload)
> > +
> > +    def add_pool(self, context, pool, driver_name):
>
>
> 我这个评论的原意是看着这两个函数是没必要添加的样子，因为代码和原有代码大同小异，能否考虑一下使用里面现有的函数来完成重新调度的过程，比如add_pool_to_agent那个rpc调用完全可以直接调用agent里面的update_pool，remove_pool_from_agent可以直接调用delete。
>
>
> 另外，当移除pool和添加pool时，是否应该考虑一下更改pool的状态，比如移除时从任意状态变为DOWN，添加时变回PENDING_UPDATE，然后交给agent，在下面添加和移除那里我会提。
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eayunstack/neutron/pull/39/files/609dd87c1c005386d538a761bbb1b4b09842b8c4#r70560261>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGUT0fBdHcpu1OfcUgqalwW7U5hV7COaks5qVFyxgaJpZM4JJO13>
> .
>